FirebaseAppDistribution/Sources/FIRAppDistribution.m

```diff
@@ -116,181 +123,160 @@ + (instancetype)appDistribution {
 
   // In the component creation block, we return an instance of `FIRAppDistribution`. Cast it and
   // return it.
+  FIRFADDebugLog(@"Instance returned: %@", instance);
   return (FIRAppDistribution *)instance;
 }
 
 - (void)signInTesterWithCompletion:(void (^)(NSError *_Nullable error))completion {
-  NSURL *issuer = [NSURL URLWithString:kIssuerURL];
-
-  [OIDAuthorizationService
-      discoverServiceConfigurationForIssuer:issuer
-                                 completion:^(OIDServiceConfiguration *_Nullable configuration,
-                                              NSError *_Nullable error) {
-                                   [self handleOauthDiscoveryCompletion:configuration
-                                                                  error:error
-                                        appDistributionSignInCompletion:completion];
-                                 }];
-}
+  FIRFADDebugLog(@"Prompting tester for sign in");
 
-- (void)signOutTester {
-  NSError *error;
-  BOOL didClearAuthState = [FIRAppDistributionAuthPersistence clearAuthState:&error];
-  // TODO (schnecle): Add in FIRLogger to report when we have failed to clear auth state
-  if (!didClearAuthState) {
-    NSLog(@"Error clearing token from keychain: %@", [error localizedDescription]);
+  if ([self isTesterSignedIn]) {
+    completion(nil);
+    return;
   }
 
-  self.authState = nil;
-  self.isTesterSignedIn = false;
-}
+  [self.UIService initializeUIState];
+  FIRInstallations *installations = [FIRInstallations installations];
 
-- (void)fetchReleases:(FIRAppDistributionUpdateCheckCompletion)completion {
-  [self.authState performActionWithFreshTokens:^(NSString *_Nonnull accessToken,
-                                                 NSString *_Nonnull idToken,
-                                                 NSError *_Nullable error) {
+  // Get a Firebase Installation ID (FID).
+  [installations installationIDWithCompletion:^(NSString *__nullable identifier,
+                                                NSError *__nullable error) {
     if (error) {
-      // TODO (schnecle): Add in FIRLogger log statement
-      NSLog(@"Error fetching fresh tokens: %@", [error localizedDescription]);
-      [self signOutTester];
+      NSString *description = error.userInfo[NSLocalizedDescriptionKey]
```

Comment:
One thing to keep in mind is I believe this completion handler is called on the main thread @maksymmalyhin can confirm. We were worried about performance / thread locking up so we put the callback on a background thread: https://github.com/firebase/firebase-ios-sdk/blob/master/Crashlytics/Crashlytics/Models/FIRCLSInstallIdentifierModel.m#L108

Might not be necessary here but just a heads up